control-plane-operator/controllers/hostedcontrolplane/pki/cert.go

```diff
@@ -34,12 +34,14 @@ func reconcileSignedCertWithAddresses(secret, ca *corev1.Secret, ownerRef config
 }
 
 func reconcileSignedCertWithKeysAndAddresses(secret *corev1.Secret, ca *corev1.Secret, ownerRef config.OwnerRef, cn, org string, usage x509.KeyUsage, extUsages []x509.ExtKeyUsage, crtKey, keyKey, caKey string, dnsNames []string, ips []string) error {
+	if secret.CreationTimestamp.IsZero() {
+		secret.Type = corev1.SecretTypeOpaque
+	}
```

Comment:
@enxebre I needed this to fix crashing, probably need to split into another PR

Reply:
same here, we might be missing something in the createOrUpdate roundtrip. FWIW  SecretTypeOpaque is default, we don't even need to explicitly set it.

Reply:
The cluster where it was reproduced is long gone, haven't investigated. In any case, the new code I have here is technically correct because that field is immutable, so not sure why not be defensive.